native/acter/src/api/reactions.rs

```diff
@@ -0,0 +1,218 @@
+                    bail!("{key} is not a reaction");
+                };
+                let room = client
+                    .core
+                    .client()
+                    .get_room(&reaction.meta.room_id)
+                    .context("Room not found")?;
+                Ok(Reaction {
+                    client: client.clone(),
+                    room,
+                    inner: reaction,
+                })
+            })
+            .await?
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Reaction {
+    client: Client,
+    room: Room,
+    inner: models::Reaction,
+}
+
+impl Deref for Reaction {
+    type Target = models::Reaction;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl Reaction {
+    pub fn event_id_str(&self) -> String {
+        self.inner.event_id().to_string()
+    }
+
+    pub fn sender(&self) -> OwnedUserId {
+        self.inner.meta.sender.clone()
+    }
+
+    pub fn origin_server_ts(&self) -> u64 {
+        self.inner.meta.origin_server_ts.get().into()
+    }
+
+    pub fn relates_to(&self) -> String {
+        self.inner.relates_to.event_id.to_string()
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ReactionManager {
+    client: Client,
+    room: Room,
+    inner: models::ReactionManager,
+}
+
+impl Deref for ReactionManager {
+    type Target = models::ReactionManager;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl ReactionManager {
+    pub(crate) fn new(
+        client: Client,
+        room: Room,
+        inner: models::ReactionManager,
+    ) -> ReactionManager {
+        ReactionManager {
+            client,
+            room,
+            inner,
+        }
+    }
+
+    pub async fn send_reaction(&self, event_id: String, key: String) -> Result<OwnedEventId> {
+        let room = self.room.clone();
+        let client = room.client();
+        let my_id = client.user_id().context("User not found")?.to_owned();
+        let event_id = EventId::parse(&event_id)?;
+
+        RUNTIME
+            .spawn(async move {
+                let member = room
+                    .get_member(&my_id)
+                    .await?
+                    .context("Couldn't find me among room members")?;
+                if !member.can_send_message(MessageLikeEventType::RoomMessage) {
+                    bail!("No permission to send message in this room");
+                }
+
+                trace!("before sending reaction");
+                let content = ReactionEventContent::new(Annotation::new(event_id, key));
+                let response = room.send(content).await?;
+
+                trace!("after sending reaction");
+                Ok(response.event_id)
+            })
+            .await?
+    }
+
+    pub async fn redact_reaction(
+        &self,
+        event_id: String,
+        reason: Option<String>,
+        txn_id: Option<String>,
+    ) -> Result<OwnedEventId> {
```

Comment:
this should figure out our own redaction without the user providing the `event_id`. The code that currently happens inline in the flutter side, should happen here.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1208/commits/87f23b9832180a2405fea636bf58679fdcfe70f2